pkg/matcher/matcher.go

```diff
@@ -39,6 +66,33 @@ func (w *allowlistMatcher) Match(fullPath string, _ os.FileInfo, _ io.ReaderAt)
 	return false, false
 }
 
+func (w *allowlistMatcher) GetCommonPrefixDirs() []string {
+	return findCommonDirPrefixes(w.allowlist)
+}
+
+func findCommonDirPrefixes(prefixes []string) []string {
+	pre := make(map[string]set.StringSet)
```

Comment:
How about prefixToSubdirsMap?

Reply:
Want to clarify that the method is doing a search in a trie-like structure finding paths with single-children nodes. Do you think replacing `find` with `search` would be better, or do you still think a "map" verb in the name would be clearer?

Reply:
I was proposing the alt name for the `pre`. I am ok with both names for the functions.

Reply:
Gotcha. Let me update it, but I will remove the `map` prefix, since golang is typed, and the decoration of the name to its type here is redundant.